bundles/org.eclipse.swt/Eclipse SWT Custom Widgets/common/org/eclipse/swt/custom/CTabFolder.java

```diff
@@ -769,6 +778,7 @@ void createItem (CTabItem item, int index) {
 	} else {
 		updateFolder(REDRAW_TABS);
 	}
+	notifyItemCountChange();
 }
 void destroyItem (CTabItem item) {
 	if (inDispose) return;
```

Comment:
Also, I'm not sure what `if (index == -1) return` is for below. How is it even possible for TabItem to not be present in parent?

Reply:
> Do you think events are not needed when TabFolder is being disposed? I have no opinion.

I think no. As a client I'm interesting if tabs are added / removed in a **living** folder instance. How many items will be / will be not in the dead instance, no one should be interested.

> Also, I'm not sure what if (index == -1) return is for below. How is it even possible for TabItem to not be present in parent?

I guess because `org.eclipse.swt.custom.CTabFolder.items` is accessible from outside of CTabFolder (horrible!) and anyone in the package can change items.